### ISSUE TITLE
Rule S2068: detect hard-coded passwords in appsettings.json and web.config files

### ISSUE THREAD

**Issue by agigleux** [SonarSource/sonar-dotnet#5427]
Follow-up of #4298 to detect hard-coded credentials in [appsettings.json](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-5.0) and [web.config](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/web-config?view=aspnetcore-6.0) files.

[S2068](https://rules.sonarsource.com/csharp/RSPEC-2068) should raised on appsettings.json or web.config files.
It should not raised on empty Password of ConnectionStrings because it is already handled by S2115.

Noncompliant
``` json
{
  "ConnectionStrings": {
    "DefaultConnection": "Server=myServerAddress;Database=myDataBase;User Id=myUsername;Password=123"
  }
}
```

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
	<connectionStrings>
		<add name="myConnection" connectionString="Server=myServerAddress;Database=myDataBase;User Id=myUsername;Password=123" /> <!-- Noncompliant -->
	</connectionStrings>
</configuration>
```

Compliant
```json
{
  "ConnectionStrings": {
    "DefaultConnection": "Server=myServerAddress;Database=myDataBase;Integrated Security=True"
  }
}
```

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
	<connectionStrings>
		<add name="myConnection" connectionString="Server=myServerAddress;Database=myDataBase;Trusted_Connection=True" />
	</connectionStrings>
</configuration>
```

**Comment by pavel-mikula-sonarsource** [comment]
It should not raise in `Web.Debug.config` and `AppSettings.Development.config` files
